### PR TITLE
Simpliy use of Configuration object emscripten.py

### PR DESCRIPTION
**Pull request by sbc100**
I'm not sure any emscripten.py has its own Configuration.  The
regular one at `shared.configuration` is already created on import
and is identical.  So this change basically reduces the number
of Configuration (and also TempFiles which configuration creates)
objects in the system.

Also rename `keywords` to `options` which is more common for
the result of opt_parse.

### CODE REVIEW

File: emscripten.py [hunk 1]
```diff
@@ -2317,49 +2309,45 @@ def _main(args):
...
+  if not options.suppressUsageWarning:
     logging.warning('''
 ==============================================================
 WARNING: You should normally never use this! Use emcc instead.
 ==============================================================
   ''')
 
   if len(positional) != 1:
-    logging.error('Must provide exactly one positional argument. Got ' + str(len(positional)) + ': "' + '", "'.join(positional) + '"')
+    logging.error('Must provide exactly one positional argument. Got ' + len(positional) + ': "' + '", "'.join(positional) + '"')
```
Comment (jgravelle-google) [inline emscripten.py]: On the python console:
```
>>> 3 + "Foo"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
```

for both python2 and python3

File: emscripten.py [hunk 2]
```diff
@@ -29,14 +29,6 @@ def path_from_root(*pathelems):
   """
   return os.path.join(__rootpath__, *pathelems)
 
-def get_configuration():
-  if hasattr(get_configuration, 'configuration'):
```
Comment (jgravelle-google) [inline emscripten.py]: wat

File: tools/shared.py [hunk 3]
```diff
@@ -834,20 +835,23 @@ def __init__(self, environ=os.environ):
...
+    CANONICAL_TEMP_DIR = self.CANONICAL_TEMP_DIR
+    TEMP_DIR = self.TEMP_DIR
+
   def get_temp_files(self):
-    return tempfiles.TempFiles(
-      tmp=self.TEMP_DIR if not self.DEBUG else get_emscripten_temp_dir(),
-      save_debug_files=os.environ.get('EMCC_DEBUG_SAVE'))
-
-def apply_configuration():
-  global configuration, DEBUG, EMSCRIPTEN_TEMP_DIR, DEBUG_CACHE, CANONICAL_TEMP_DIR, TEMP_DIR
```
Comment (jgravelle-google) [inline tools/shared.py]: So I don't see a `configuration` declared anywhere else in `shared`.
I didn't know `global` worked that way. That's kinda spooky.

File: emscripten.py (hunk 2 above)
Comment (kripken) [reply emscripten.py]: yeah, i never really understood this part of the code myself

**Comment by sbc100** [comment]
Abandoning for now.  Will revisit later perhaps.

**Pull request closed without merging**